x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/details/index.tsx

```diff
@@ -39,198 +41,261 @@ import { getStepsData } from '../helpers';
+              defaultMessage="Created by: {by} on {date}"
               values={{
-                by: rule?.updated_by ?? i18n.UNKNOWN,
+                by: rule?.created_by ?? i18n.UNKNOWN,
                 date: (
                   <FormattedDate
-                    value={rule?.updated_at ?? new Date().toISOString()}
-                    fieldName="updatedAt"
+                    value={rule?.created_at ?? new Date().toISOString()}
+                    fieldName="createdAt"
                   />
                 ),
               }}
-            />
-          ) : (
-            ''
-          ),
-        ]
-      ),
-    [loading, rule]
-  );
-
-  const signalDefaultFilters = useMemo(
-    () => (ruleId != null ? buildSignalsRuleIdFilter(ruleId) : []),
-    [ruleId]
-  );
-  return (
-    <>
-      <WithSource sourceId="default">
-        {({ indicesExist, indexPattern }) => {
-          return indicesExistOrDataTemporarilyUnavailable(indicesExist) ? (
-            <GlobalTime>
-              {({ to, from }) => (
-                <StickyContainer>
-                  <FiltersGlobal>
-                    <SiemSearchBar id="global" indexPattern={indexPattern} />
-                  </FiltersGlobal>
-
-                  <WrapperPage>
-                    <HeaderPage
-                      backOptions={{
-                        href: `#${DETECTION_ENGINE_PAGE_NAME}/rules`,
-                        text: i18n.BACK_TO_RULES,
-                      }}
-                      badgeOptions={{ text: i18n.EXPERIMENTAL }}
-                      border
-                      subtitle={subTitle}
-                      subtitle2={[
-                        lastSignals != null ? (
-                          <>
-                            {detectionI18n.LAST_SIGNAL}
-                            {': '}
-                            {lastSignals}
-                          </>
-                        ) : null,
-                        'Status: Comming Soon',
-                      ]}
-                      title={title}
-                    >
-                      <EuiFlexGroup alignItems="center">
-                        <EuiFlexItem grow={false}>
-                          <RuleSwitch
-                            id={rule?.id ?? '-1'}
-                            enabled={rule?.enabled ?? false}
-                            optionLabel={i18n.ACTIVATE_RULE}
-                          />
-                        </EuiFlexItem>
+            />,
+            rule?.updated_by != null ? (
+              <FormattedMessage
+                id="xpack.siem.detectionEngine.ruleDetails.ruleUpdateDescription"
+                defaultMessage="Updated by: {by} on {date}"
+                values={{
+                  by: rule?.updated_by ?? i18n.UNKNOWN,
+                  date: (
+                    <FormattedDate
+                      value={rule?.updated_at ?? new Date().toISOString()}
```

Comment:
From the backend we always provide a `updated_at` and a `created_at` without mistake as far as I know now. So the checks for these should never happen.

However, same issues with TypeScript saying that these two can never be null/undefined. If you want to be safe and just guard against a regression bug on our side or something else, I would change the type system to reflect that and also I would not put "fake dates" such as `new Date().toISOString()` but rather defer to using a `--` nonexistent placeholder to show that they just aren't present.

Up to you.

Reply:
cc @XavierM -- this may be an artifact from initial development when they weren't always present?